app.js

```diff
@@ -6,23 +6,33 @@ const {
     getEndpoints, 
     getArticles, 
     getArticleById,
-    getCommentsById
+    getCommentsById,
+    postCommentsById
 } = require("./controllers/news.controllers.js")
 
+app.use(express.json());
+
 app.get('/api', getEndpoints)
 app.get('/api/topics', getTopics)
 app.get('/api/articles', getArticles)
 app.get('/api/articles/:article_id', getArticleById)
 app.get('/api/articles/:article_id/comments', getCommentsById)
 
+app.post('/api/articles/:article_id/comments', postCommentsById)
 
 app.all('*', (req,res)=> res.status(404).send({msg : 'Not Found'}));
 
 app.use((err, req, res, next) => {
-    if (err.code === '42703') {
+    if (err.code === '22P02') {
+        res.status(400).send({msg: 'Bad Request'});
+    } else if (err.code === '42703') {
         res.status(400).send({msg: 'Bad Request'});
     } else if (err.code === '42601') {
         res.status(400).send({msg: 'Bad Request'});
+    } else if (err.code === '23502') {
+        res.status(400).send({msg: 'Bad Request'});
+    } else if (err.code === '23503') {
+        res.status(404).send({msg: 'Not Found'});
     } else  next(err);
```

Comment:
^^ I think at this point you should start to consider creating an errors.js file to contain your error handlers :)